### PR TITLE
Extract gog mehrdad

### PR DESCRIPTION
**Pull request by mehrdadhalali**
This is a MVP of the GOG extractor. TODO:
- Refactor code
- Add tests
- Make more efficient (current running time: about 80s)

### CODE REVIEW

File: web_scraping/gog_scraper/README.md [hunk 1]
```diff
@@ -1 +1,22 @@
-# GOG Scraper
\ No newline at end of file
+# 🚀 GOG Scraper
+
+This folder contains scripts to **scrape** the latest games from [the GOG website.](https://www.gog.com/en/games?releaseStatuses=new-arrival&order=desc:releaseDate&hideDLCs=true)
+
+---
+
+## 🛠️ Prerequisites
```
Comment (Leon-JS02) [inline web_scraping/gog_scraper/README.md:7]: Cool emojis. This file can always be added to as this folder develops.

File: web_scraping/gog_scraper/scrape_gog.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+"""Script to scrape the latest games from GOG."""
+
+from time import perf_counter
+from datetime import datetime
+from bs4 import BeautifulSoup
```
Comment (Leon-JS02) [inline web_scraping/gog_scraper/scrape_gog.py:5]: A line should be left separating different types of imports.

File: web_scraping/gog_scraper/scrape_gog.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    html = get_html(url)
+
+    game_urls = get_game_urls_from_page(html)
+
+    game_details = list(map(get_game_data_from_url, game_urls))
+
+    day_str = datetime.strftime(day, "%d/%m/%Y")
+
+    relevant_games = [game for game in game_details
```
Comment (Leon-JS02) [inline web_scraping/gog_scraper/scrape_gog.py:34]: Snazzy comprehension.

File: web_scraping/gog_scraper/scrape_gog_game.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    html = html_bytes.decode("utf_8")
+    return html
+
+
+def format_os(os_string: str) -> str:
+    """Turns the OS into the proper format."""
+
+    os_map = {
+        "windows": "Windows",
+        "mac": "Mac",
```
Comment (Leon-JS02) [inline web_scraping/gog_scraper/scrape_gog_game.py:22]: Is this not just calling `str.title()`?

File: web_scraping/gog_scraper/scrape_gog_game.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            genres = [link.text for link in genre_links]
+
+        elif row_label == "Tags:":
+            tag_links = row.find_all("a")
+            tags = [link.text for link in tag_links]
+
+    rating_rows = soup.find_all(
+        "div", {"class": "table__row details__rating details__row"})
+
+    for row in rating_rows:
```
Comment (Leon-JS02) [inline web_scraping/gog_scraper/scrape_gog_game.py:77]: Could some of this logic be separated into smaller functions?

**Review by Leon-JS02 — DISMISSED** [review]
Good functional code for an MVP 🫶  
With some small refactors this will be more solid and less complex.

File: web_scraping/gog_scraper/scrape_gog_game.py (hunk 5 above)
Comment (mehrdadhalali) [reply web_scraping/gog_scraper/scrape_gog_game.py:77]: In progress!

File: web_scraping/gog_scraper/scrape_gog_game.py (hunk 4 above)
Comment (mehrdadhalali) [reply web_scraping/gog_scraper/scrape_gog_game.py:22]: What do you mean

Comment (Leon-JS02) [reply web_scraping/gog_scraper/scrape_gog_game.py:22]: This dictionary is mapping platform names to capitalised platform names.

`'windows'.title() # Windows`

**Review by MichaelHassan — COMMENTED** [review]
Leons beaten me to the comments. Looks good but get_game_data_from_url function needs to be broken up. Once changed happy to come back and approve

File: web_scraping/gog_scraper/scrape_gog_game.py (hunk 4 above)
Comment (mehrdadhalali) [reply web_scraping/gog_scraper/scrape_gog_game.py:22]: Oh yeah you're right.

File: web_scraping/gog_scraper/scrape_gog_game.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+def format_price(price_str: str) -> int:
+    """Formats a price's string."""
+
+    if "free" in price_str.lower():
+        return 0
+
+    return int(float(price_str)*100)
+
+
+def find_price(price_div) -> int:
```
Comment (Leon-JS02) [inline web_scraping/gog_scraper/scrape_gog_game.py:27]: These are nice functions! Well divided.

**Review by Leon-JS02 — DISMISSED** [review]
A nice refactor, and handy features added.

**Review by MichaelHassan — APPROVED** [review]
Thank you for compartmentalising the get_game_data_from_url function

**Review by Leon-JS02 — APPROVED** [review]
This is neat code.

**Pull request merged**